src/stan_math_backend/Lower_functions.ml

```diff
@@ -316,7 +316,9 @@ let lower_standalone_fun_def namespace_fun
   let return_type, return_stmt =
     match fdrt with
     | Void -> (Void, fun e -> Expression e)
-    | _ -> (Auto, fun e -> Return (Some e)) in
+    | ReturnType ut when UnsizedType.is_int_type ut ->
+        (lower_type ut Int, fun e -> Return (Some e))
+    | ReturnType ut -> (lower_type ut Double, fun e -> Return (Some e)) in
```

Comment:
Special-casing on `is_int_type` is not necessary; `lower_type` ignores the `scalar` argument when the inner type is integer.